packages/react/src/primitives/PhoneNumberField/PhoneNumberField.tsx

```diff
@@ -6,42 +6,71 @@ import { CountryCodeSelect } from './CountryCodeSelect';
 import { PhoneNumberFieldProps, Primitive } from '../types';
 import { ComponentText } from '../shared/constants';
 import { TextField } from '../TextField';
+import { useDeprecationWarning } from '../../hooks/useDeprecationWarning';
 
 const PhoneNumberFieldPrimitive: Primitive<PhoneNumberFieldProps, 'input'> = (
   {
     autoComplete = 'tel-national',
     className,
     countryCodeName,
-    countryCodeLabel = ComponentText.PhoneNumberField.countryCodeLabel,
+    dialCodeName,
+    countryCodeLabel = ComponentText.PhoneNumberField.dialCodeLabel,
+    dialCodeLabel = ComponentText.PhoneNumberField.dialCodeLabel,
     defaultCountryCode,
+    defaultDialCode,
     hasError,
     isDisabled,
     isReadOnly,
     onCountryCodeChange,
+    onDialCodeChange,
     onInput,
     size,
     type,
     variation,
     countryCodeRef,
+    dialCodeRef,
     dialCodeList,
     ...rest
   },
   ref
 ) => {
+  /**
+   * Merge all dial/country code values in preparation of countryCode values being removed preferring dial code values
+   */
+  dialCodeName = dialCodeName || countryCodeName;
+  dialCodeLabel = dialCodeLabel || countryCodeLabel;
+  defaultDialCode = defaultDialCode || defaultCountryCode;
+  onDialCodeChange = onDialCodeChange || onCountryCodeChange;
+  dialCodeRef = dialCodeRef || countryCodeRef;
+
+  const deprecationMessage =
+    'PhoneNumberField countryCodeName, countryCodeLabel, defaultCountryCode, onCountryCodeChange, and countryCodeRef prop will be deprecated in next major release of @aws-amplify/ui-react. Please use dialCodeName, dialCodeLabel, defaultDialCode, onDialCodeChange, and dialCodeRef instead.';
```

Comment:
Could probably use some more tweaking but suggest updating to:
```suggestion
    'The PhoneNumberField component props: countryCodeName, countryCodeLabel, defaultCountryCode, onCountryCodeChange, and countryCodeRef props are deprecated and will be removed in the next major release of @aws-amplify/ui-react. Please update to dialCodeName, dialCodeLabel, defaultDialCode, onDialCodeChange, and dialCodeRef respectively.';
```